elastic-agent-client.proto

```diff
@@ -189,12 +262,29 @@ message UnitExpected {
     uint64 config_state_idx = 4;
     // Resulting configuration. (If the application already has the current `config_state_idx` this
     // will be empty.)
-    string config = 5;
+    UnitExpectedConfig config = 5;
+    // Log level of the unit.
+    UnitLogLevel log_level = 6;
+}
+
+// Only provided on first checkin expected response to the component.
+//
+// Includes the agent information that the component might want to use for its events.
+message CheckinAgentInfo {
+    // ID is the Elastic Agent's unique ID.
+    string id = 1;
+    // Version is the version of the running Elastic Agent.
+    string version = 2;
```

Comment:
Is it worth having this be a struct with major.minor.patch fields?

Reply:
Both Elastic Agent and Fleet Server just use version as a string. I think doing something like that is over the top.

Reply:
I would keep it as a string, IIRC we a type that could encapsulate the version and allow us to do any kind of comparison.